crates/wasmtime/src/func.rs

```diff
@@ -729,13 +730,18 @@ impl fmt::Debug for Func {
     }
 }
 
-pub(crate) fn catch_traps(
+pub(crate) fn invoke_wasm_and_catch_traps(
     vmctx: *mut VMContext,
     store: &Store,
     closure: impl FnMut(),
 ) -> Result<(), Trap> {
     let signalhandler = store.signal_handler();
     unsafe {
+        let canary = 0;
+        let _auto_reset_canary = store
+            .externref_activations_table()
+            .set_stack_canary(&canary);
```

Comment:
It does work today (the tests would fail if it didn't, since they are checking the reference counts and asserting that they go down after GC, which means we did sweep the table, which means we saw the canary).

I agree that it is slightly fragile. I don't know what we could do to avoid it without something like `test::blackbox` or calling `getcontext` to get the SP (eww out-of-line call).